pkg/validation/options.go

```diff
@@ -282,6 +282,13 @@ func parseProviderInfo(o *options.Options, msgs []string) []string {
 	case *providers.GitLabProvider:
 		p.AllowUnverifiedEmail = o.InsecureOIDCAllowUnverifiedEmail
 		p.Groups = o.GitLabGroup
+		err := p.AddProjects(o.GitlabProjects)
+		if err != nil {
+			msgs = append(msgs, "failed to setup gitlab project access level")
+		}
+		p.SetAllowedGroups(p.PrefixAllowedGroups())
+		p.SetProjectScope()
+		p.EmailDomains = o.EmailDomains
```

Comment:
Why was this added? I don't see it used in your changes, I only see it set.

You just need to be sure to set a `session.Email`. Authorizing the email against the `--email-domains` or `--authenticated-emails-file` is the responsibility of `oauthproxy.go` based on how providers populate that mandatory field. It handles the complex logic dealing with `*` support.

So I think this shouldn't be here.